osquery/tables/system/darwin/system_info.cpp

```diff
@@ -0,0 +1,81 @@
+/*
+ *  Copyright (c) 2014, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under the BSD-style license found in the
+ *  LICENSE file in the root directory of this source tree. An additional grant
+ *  of patent rights can be found in the PATENTS file in the same directory.
+ *
+ */
+
+#include <osquery/database.h>
+#include <osquery/tables.h>
+#include "osquery/core/conversions.h"
+
+#include <CoreFoundation/CoreFoundation.h>
+#include <IOKit/IOKitLib.h>
+
+namespace osquery {
+namespace tables {
+
+Status getCpuSerial(std::string& cpu_serial) {
+  auto matching = IOServiceMatching("IOPlatformExpertDevice");
```

Comment:
If you did something like:
```cpp
static std::string serial_cache;
if (!serial_cache.empty()) {
  cpu_serial = serial_cache;
  return;
}
```

and then wrote:

```cpp
cpu_serial = serial_cache = stringFromCFString(serialNumber);
```

below. You would only ever run the IOKit and CF related methods once.This is appropriate as the CPU serial will not change. :)